Investigate 3 partests sometimes failing on weekly

* `run/bitsets.scala`
* `neg/t8266-invalid-interp.scala`
* `neg/t7848-interp-warn.scala`

They don't emit the same warnings as what they're supposed to.

I see https://scala-webapps.epfl.ch/jenkins/job/scalajs-task-worker/59616/console
```
!! 817 - run/bitsets.scala                         [output differs]
% diff /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.5/test/files/run/bitsets-run.log /localhome/jenkins/a/workspace/scalajs-task-worker_2/partest/fetchedSources/2.11.5/test/files/run/bitsets.check
@@ -1,4 +1,3 @@
-warning: there were three deprecation warnings; re-run with -deprecation for details
 ms0 = BitSet(2)
 ms1 = BitSet(2)
 ms2 = BitSet(2)
```
which is a real deprecation
```
test/files/run/bitsets.scala:38: warning: method toImmutable in class BitSet is deprecated: If this BitSet contains a value that is 128 or greater, the result of this method is an 'immutable' BitSet that shares state with this mutable BitSet. Thus, if the mutable BitSet is modified, it will violate the immutability of the result.
  Console.println("mi0 = " + ms0.toImmutable)
```
The t8266 message was changed in 2.11.4. I don't see the test error, though.

--- a/project/Build.scala
+++ b/project/Build.scala
@@ -1384,6 +1384,9 @@ object Build extends sbt.Build {
           fork in Test := true,
           javaOptions in Test += "-Xmx1G",
 
+          // Override the dependency of partest - see #1889
+          dependencyOverrides += "org.scala-lang" % "scala-library" % scalaVersion.value % "test",
+
           testFrameworks ++= {
             if (shouldPartest.value)
               Seq(new TestFramework("scala.tools.partest.scalajs.Framework"))

